### PR TITLE
Add sleep conditionally (RhBug:1741931)

### PR DESCRIPTION
**Pull request by j-mracek**
Remove sleep step,  retries download conditionally, and additional retries for mirrorlist and metalink

### CODE REVIEW

**Comment by j-mracek** [comment]
Related bug: https://bugzilla.redhat.com/show_bug.cgi?id=1756400

File: librepo/downloader.c [hunk 1]
```diff
@@ -704,23 +711,22 @@ select_suitable_mirror(LrDownload *dd,
...
-            sleep(mirrors_iterated);
-        }
         // Iterate over mirror for the target
         for (GSList *elem = target->lrmirrors; elem; elem = g_slist_next(elem)) {
             LrMirror *c_mirror = elem->data;
             gchar *mirrorurl = c_mirror->mirror->url;
 
             // first iteration, filter out mirrors that failed previously
             if (mirrors_iterated == 0) {
+                if (c_mirror->mirror->protocol != LR_PROTOCOL_FILE)
```
Comment (jrohel) [inline librepo/downloader.c]: So, if one of mirror is not local then reiteration is enabled. But then local paths are retried too, isn't it? You wrote "retry local paths nave no reason" above. I think you forgot add condition code at line 743. 
```
            } else if (c_mirror->mirror->protocol == LR_PROTOCOL_FILE) {
                continue;
```
Or am I misunderstand the code?

File: librepo/downloader.c [hunk 2]
```diff
@@ -302,10 +302,17 @@ static gboolean
...
-can_retry_download(const LrDownload *download, int num_of_tried_mirrors, gboolean complete_path_or_baseurl)
+can_retry_download(const LrDownload *download, int num_of_tried_mirrors,
+                   const char * complete_path_or_baseurl)
 {
     if (complete_path_or_baseurl) {
-        return download->allowed_mirror_failures > num_of_tried_mirrors;
+        if (g_str_has_prefix(complete_path_or_baseurl, "file:/")) {
+            return FALSE;
+        }
+        if (download->allowed_mirror_failures > num_of_tried_mirrors) {
```
Comment (jrohel) [inline librepo/downloader.c]: Next 4 lines do the same as the original one line code.

**Comment by jrohel** [comment]
@rh-atomic-bot r+

**Pull request merged**